packages/yarnpkg-core/sources/formatUtils.ts

```diff
@@ -400,7 +400,13 @@ export function addLogFilterSupport(report: Report, {configuration}: {configurat
       return defaultLevel;
 
     if (logFiltersByText.size > 0) {
-      const level = logFiltersByText.get(stripAnsi(text));
+      let level;
+      for (const [filterText, filterLevel] of logFiltersByText.entries()) {
+        if ((new RegExp(filterText)).test(stripAnsi(text))) {
```

Comment:
The RegExp should be precompiled, as otherwise we incur the cost for every output line, which can be a lot (for example during fetching, one line per fetch).

Reply:
Good catch!
I fixed this issue with the rework to use Globs instead of RegExps.